ProteoformSuiteInternal/ProteoformCommunity.cs

```diff
@@ -25,71 +22,89 @@ public bool has_e_and_t_proteoforms
+                        ProteoformRelation best_relation = pf1.candidate_relatives
+                            .Select(pf2 => new ProteoformRelation(pf1, pf2, relation_type, pf1.modified_mass - pf2.modified_mass))
+                            .Where(r => r.candidate_ptmset != null) // don't consider unassignable relations for ET
+                            .OrderBy(r => r.candidate_ptmset.ptm_rank_sum) // get the best explanation for the experimental observation
+                            .FirstOrDefault();
+
+                        pf1.candidate_relatives = best_relation != null ?
+                            new List<Proteoform> { best_relation.connected_proteoforms[1] } : 
+                            new List<Proteoform>();
+                    }
+                }
             });
 
-            List<ProteoformRelation> relations =
+            IEnumerable<ProteoformRelation> relations =
                 (from pf1 in pfs1
                  from pf2 in pf1.candidate_relatives
-                 select new ProteoformRelation(pf1, pf2, relation_type, pf1.modified_mass - pf2.modified_mass))
-                 .OrderBy(r => r.delta_mass).ToList();
+                 select new ProteoformRelation(pf1, pf2, relation_type, pf1.modified_mass - pf2.modified_mass));
 
-            return count_nearby_relations(relations);  //putative counts include no-mans land
+            return count_nearby_relations(relations.ToList());  //putative counts include no-mans land
         }
 
-        public bool allowed_ee_relation(ExperimentalProteoform pf1, ExperimentalProteoform pf2)
+        public bool allowed_relation(Proteoform pf1, Proteoform pf2, ProteoformComparison relation_type)
         {
-            return pf1.modified_mass >= pf2.modified_mass
-                && pf1 != pf2
-                && (!Lollipop.neucode_labeled || pf1.lysine_count == pf2.lysine_count)
-                && pf1.modified_mass - pf2.modified_mass <= Lollipop.ee_max_mass_difference
-                && Math.Abs(pf1.agg_rt - pf2.agg_rt) <= Lollipop.ee_max_RetentionTime_difference;
-            //where ProteoformRelation.mass_difference_is_outside_no_mans_land(pf1.modified_mass - pf2.modified_mass)
-            //putative counts include no-mans land, currently
+            switch (relation_type)
+            {
+                case (ProteoformComparison.ExperimentalTheoretical):
+                case (ProteoformComparison.ExperimentalDecoy):
+                    return (!Lollipop.neucode_labeled || pf2.lysine_count == pf1.lysine_count)
+                        && (pf1.modified_mass - pf2.modified_mass) >= Lollipop.et_low_mass_difference
+                        && (pf1.modified_mass - pf2.modified_mass) <= Lollipop.et_high_mass_difference
+                        && (pf2.ptm_set.ptm_combination.Count < 3 || pf2.ptm_set.ptm_combination.Select(ptm => ptm.modification.monoisotopicMass).All(x => x == pf2.ptm_set.ptm_combination.First().modification.monoisotopicMass));
+
```

Comment:
This bit of code determines which ET pairs are valid, not which theoretical proteoforms are generated. I'm only considering connections with Theoreticals with more than 3 PTMs if they are the same PTM.